src/eth_provider/provider.rs

```diff
@@ -765,10 +772,7 @@ where
 
     /// Get a header from the database based on the filter.
     async fn header(&self, id: BlockHashOrNumber) -> EthProviderResult<Option<StoredHeader>> {
-        let filter = match id {
```

Comment:
This doesn't work, id_to_filter doesn't return the correct filter

Reply:
My bad, will take that out